### ISSUE TITLE
Error from mycroft core when using 'number' type in settings

### ISSUE THREAD

**Issue by richhowley** [MycroftAI/mycroft-core#2298]
I am developing a skill that uses the "number" type for two settings.  When the skill is loaded (Pixroft or Linux dev) an error comes up constantly in the CLI

```
Traceback (most recent call last):
  File "/home/pi/mycroft-core/mycroft/skills/settings.py", line 411, in _poll_skill_settings
    self.update_remote()
  File "/home/pi/mycroft-core/mycroft/skills/settings.py", line 370, in update_remote
    skills_settings = self._request_my_settings(self.skill_gid)
  File "/home/pi/mycroft-core/mycroft/skills/settings.py", line 518, in _request_my_settings
    self._type_cast(skill_settings, to_platform='core')
  File "/home/pi/mycroft-core/mycroft/skills/settings.py", line 492, in _type_cast
    if "." in value:
TypeError: argument of type 'int' is not iterable

```
I get the same error if I do this:

```
>>> value = 4
>>> if "." in value:
...     pirnt("dot")
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'int' is not iterable
```
It looks like the error is because the variable 'value' is already an int when checking for ".".  I confirmed this by doing:
```
                    if to_platform == 'core':
                        print("*************************************** settings") 
                        print(type(value))
                        if "." in str(value):
```

and seeing:

```
*************************************** settings
<class 'int'>
```

in the terminal.  

This change:
```
                        if "." in str(value):
                            sections[i]['fields'][j]['value'] = float(value)
                        else:
                            sections[i]['fields'][j]['value'] = int(value)
```
cleaned up the errors.

**Comment by ChanceNCounter** [comment]
Looks like that fix was pushed to dev in 3cacdaa.

**Comment by krisgesling** [comment]
Yep should be fixed by #2258 

So will close this issue, but please let us know if you spot this after the release of 19.08.
